lib/core/msdf-sys/src/lib.rs

```diff
@@ -0,0 +1,163 @@
+extern crate wasm_bindgen;
+#[macro_use] extern crate slice_as_array;
+
+use wasm_bindgen::prelude::*;
+use js_sys::*;
+
+const MAX_MSDF_SIZE : usize = 32;
+pub const MSDF_CHANNELS_COUNT : usize = 3;
+const MSDF_DATA_SIZE : usize = MAX_MSDF_SIZE*MAX_MSDF_SIZE*MSDF_CHANNELS_COUNT;
+
+pub struct FontHandle {
+    handle: JsValue
+}
+
+pub struct MSDF {
+    pub width  : usize,
+    pub height : usize,
+    pub data   : [f32;MSDF_DATA_SIZE]
+}
+
+pub struct MSDFParameters {
+    width                         : usize,
+    height                        : usize,
+    edge_coloring_angle_threshold : f64,
+    range                         : f64,
+    scale                         : (f64, f64),
+    translate                     : (f64, f64),
+    edge_threshold                : f64,
+    overlap_support               : bool
+}
+
+#[wasm_bindgen(module = "msdfgen_wasm.js")]
+extern {
+
+fn ccall(
+    name        : &str,
+    return_type : &str,
+    types       : js_sys::Array,
+    values      : js_sys::Array
+) -> JsValue;
+
+fn getValue(address: usize, a_type: &str) -> JsValue;
+
+fn _msdfgen_maxMSDFSize() -> usize;
+
+fn _msdfgen_generateMSDF(
+    width                           : usize,
+    height                          : usize,
+    font_handle                     : JsValue,
+    unicode                         : u32,
+    edge_coloring_angle_threshold   : f64,
+    range                           : f64,
+    scale_x                         : f64,
+    scale_y                         : f64,
+    translate_x                     : f64,
+    translate_y                     : f64,
+    edge_threshold                  : f64,
+    overlap_support                 : bool
+) -> usize;
+
+fn _msdfgen_freeFont(font_handle: JsValue);
+
+}
+
+fn make_js_array_from_data_view(data: &[u8]) -> Uint8Array {
+    unsafe {
+        return Uint8Array::view(data)
+    }
+}
+
+pub fn load_font_memory(data: &[u8]) -> FontHandle {
+
+    let param_types = js_sys::Array::new_with_length(2);
+    param_types.set(0, JsValue::from_str("array"));
+    param_types.set(1, JsValue::from_str("number"));
+    let params = js_sys::Array::new_with_length(2);
+    params.set(0, JsValue::from(make_js_array_from_data_view(data)));
+    params.set(1, JsValue::from_f64(data.len() as f64));
+    let handle = ccall(
+        "msdfgen_loadFontMemory",
+        "number",
+        param_types,
+        params);
+    FontHandle { handle }
+
+}
+
+impl Drop for FontHandle {
+    fn drop(&mut self) {
+        _msdfgen_freeFont(self.handle.clone())
+    }
+}
```

Comment:
This part of the code should be defined close to `FontHandle` as it implements its crucial functionality. Basically, we should define structs, types, and associated impls next to each other. Such blocks should be separated with visual sections:
```
// ==================
// === FontHandle ===
// ==================

// struct def
// related type defs
// impls

// =====================
// === SomethingElse ===
// =====================